### PR TITLE
Unicode/char32 t

### PR DESCRIPTION
**Pull request by theodelrieu**
Following up #634

### CODE REVIEW

**Comment by theodelrieu** [comment]
@djowel Time for Windows now, unfortunately I don't have one right now, do you have an idea on why it fails with 2017 only?

**Comment by djowel** [comment]
> @djowel Time for Windows now, unfortunately I don't have one right now, do you have an idea on why it fails with 2017 only?

The other fails too (2015, 2017 and 2019). Type mismatch, it seems.

**Comment by theodelrieu** [comment]
>The other fails too (2015, 2017 and 2019).

Indeed... I should open my eyes before posting.

Anyway, I'll try to get it working tonight.

**Comment by theodelrieu** [comment]
@djowel This is because of the `__cplusplus`, MSVC returns `199711` by default. Is there an existing macro within boost (e.g. `BOOST_HAS_CXX11` or something) that deals with this kind of things already?

**Comment by djowel** [comment]
merged

File: include/boost/spirit/home/x3/support/traits/print_attribute.hpp [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include <boost/fusion/include/for_each.hpp>
 #include <boost/spirit/home/x3/support/traits/attribute_category.hpp>
 #include <boost/spirit/home/x3/support/traits/is_variant.hpp>
+#include <boost/spirit/home/support/char_encoding/unicode.hpp>
```
Comment (Kojoley) [inline include/boost/spirit/home/x3/support/traits/print_attribute.hpp:17]: I blazed through changes for the release and stuck on this line. It unconditionally includes the header which we previously only included when `BOOST_SPIRIT_X3_UNICODE` is defined.

Comment (djowel) [reply include/boost/spirit/home/x3/support/traits/print_attribute.hpp:17]: Oh. That slipped through. We'll need to fix this for the 1.76.0 Beta candidate. @theodelrieu ?

Comment (Kojoley) [reply include/boost/spirit/home/x3/support/traits/print_attribute.hpp:17]: I already fixed it in develop. Beta is already out, and master branch is closed, but we will have an opportunity to merge it for the release later on.

**Pull request merged**